src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/MetaService.cs

```diff
@@ -108,9 +130,16 @@ private LegendOptionsMetaValueModel<IEnumerable<TimePeriodMetaViewModel>> GetTim
         }
 
         private Dictionary<string, LegendOptionsMetaValueModel<Dictionary<string,
-            LabelOptionsMetaValueModel<IEnumerable<LabelValueViewModel>>>>> GetFilters(long subjectId)
+            LabelOptionsMetaValueModel<IEnumerable<LabelValueViewModel>>>>> GetFilters(
+            long subjectId,
+            SubjectMetaQueryContext query = null)
         {
-            return _filterService.GetFiltersBySubjectId(subjectId).ToDictionary(
+            return _filterService.GetFilters(subjectId,
+                GetYears(query),
```

Comment:
It feels like it would be better to pass through the entire query context rather than individual parameters. This would probably be more sustainable as we start to add more location levels as well.

Reply:
I already did what you suggested throughout the Data API project where the query context class exists. The call that you pointed out is to a database service classes living in the model project which know nothing about SubjectMetaQueryContext. I think given the number of parameters, it makes sense to move it, so I will do. :+1:

Reply:
Cool, same applies for `IObservationService` too. Should clean things up quite nicely :smile:

Reply:
That's done now.